src/FixedRateIrm.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {IIrm} from "../lib/morpho-blue/src/interfaces/IIrm.sol";
+import {IFixedRateIrm} from "./interfaces/IFixedRateIrm.sol";
+
+import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import {Id, MarketParams, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+/// @title FixedRateIrm
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+contract FixedRateIrm is IFixedRateIrm {
+    using MarketParamsLib for MarketParams;
+
+    /* EVENTS */
+
+    /// @notice Emitted when a borrow rate is set.
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    /* ERRORS */
+
+    /// @dev Thrown when the rate is not already set for this market.
+    string public constant RATE_NOT_SET = "rate not set";
+    /// @dev Thrown when the rate is already set for this market.
+    string public constant RATE_SET = "rate set";
+    /// @dev Thrown when trying to set the rate at zero.
+    string public constant RATE_ZERO = "rate zero";
+
+    /* STORAGE */
+
+    /// @notice Borrow rates.
+    mapping(Id => uint256) public _borrowRate;
+
+    /* SETTER */
+
+    /// @inheritdoc IFixedRateIrm
+    function setBorrowRate(Id id, uint256 newBorrowRate) external {
```

Comment:
quid of the proposal of having an owner that can update the rate later on? It seems a legit usecase.

Reply:
seems like an other IRM to me

Reply:
yes i'm worried about whether we have the correct spec for this one. Like we did not ask if integrators would use that IRM at all. I think we should do it

Reply:
Why has this conversation been resolved ? Do we have more info from integrators ?

Reply:
I'm interested in this discussion as well lmao. Why is it resolved?
I also feel like the feature of having the first owner able to change the rate or transfer ownership is interesting; and I also agree it's another IRM. I'd like to be sure this IRM is the one to be prioritized over the other though

Reply:
You could also have one irm per rate (also solves rate squatting). Changing the rate could be implemented by reallocating at the metamorpho level.

Reply:
> You could also have one irm per rate (also solves rate squatting). Changing the rate could be implemented by reallocating at the metamorpho level.

This adds more operational cost though as you need to whitelist each IRM